drivers/display/display_ist3931.c

```diff
@@ -0,0 +1,289 @@
+	ist3931_driver_set_display_on(dev);
+	ist3931_driver_set_com_pad_map(dev);
+	return 0;
+}
+
+static int ist3931_init(const struct device *dev)
+{
+	const struct ist3931_config *config = dev->config;
+
+	if (!config->bus_ready(dev)) {
+		LOG_ERR("I2C device not ready");
+		return -ENODEV;
+	}
+	LOG_INF("I2C device ready");
+	if (config->reset_gpio.port != NULL) {
+		if (!gpio_is_ready_dt(&config->reset_gpio)) {
+			LOG_ERR("Reset GPIO device not ready");
+			return -ENODEV;
+		}
+		if (gpio_pin_configure_dt(&config->reset_gpio, GPIO_OUTPUT_INACTIVE)) {
+			LOG_ERR("Couldn't configure reset pin");
+			return -EIO;
+		}
+
+		if (ist3931_init_device(dev)) {
+			LOG_ERR("Failed to initialize device!");
+			return -EIO;
+		}
+
+		return 0;
+	}
+	return 0;
+}
+
+static void ist3931_get_capabilities(const struct device *dev, struct display_capabilities *caps)
+{
+	const struct ist3931_config *config = dev->config;
+
+	memset(caps, 0, sizeof(struct display_capabilities));
+	caps->x_resolution = config->width;
+	caps->y_resolution = config->height;
+	caps->supported_pixel_formats = PIXEL_FORMAT_MONO10 | PIXEL_FORMAT_MONO01;
+	caps->current_pixel_format = PIXEL_FORMAT_MONO01;
+	caps->current_orientation = DISPLAY_ORIENTATION_NORMAL;
+}
+
+static int ist3931_write(const struct device *dev, const uint16_t x, const uint16_t y,
+			 const struct display_buffer_descriptor *desc, const void *buf)
+{
+	uint8_t *write_data_start = (uint8_t *)buf;
+	int ret;
+
+	ret = ist3931_driver_set_ay(dev, y);
+	if (ret < 0) {
+		return ret;
+	}
+	ret = ist3931_driver_set_ax(dev, y);
+	if (ret < 0) {
+		return ret;
+	}
+
+	__ASSERT(x <= IST3931_RAM_WIDTH, "x is out of range");
+	__ASSERT(y <= IST3931_RAM_HEIGHT, "y is out of range");
+	__ASSERT(x + desc->width <= IST3931_RAM_WIDTH, "x+width is out of range");
+	__ASSERT(y + desc->height <= IST3931_RAM_HEIGHT, "y+height is out of range");
+
+	for (uint16_t i = 0; i < desc->height; i++) {
+		ist3931_driver_set_ay(dev, i + y);
+		ist3931_driver_set_ax(dev, x);
+		ist3931_write_bus(dev, write_data_start, 0, desc->width / 8);
+		write_data_start += desc->width / 8;
+	}
+	return 0;
+}
+
+static inline int ist3931_blanking_on(const struct device *dev)
+{
+	return ist3931_driver_sleep_on_off(dev, false);
+}
+
+static inline int ist3931_blanking_off(const struct device *dev)
+{
+	return ist3931_driver_sleep_on_off(dev, true);
+}
+
+static const struct display_driver_api ist3931_api = {
+	.write = ist3931_write,
+	.get_capabilities = ist3931_get_capabilities,
+	.blanking_on = ist3931_blanking_on,
+	.blanking_off = ist3931_blanking_off,
+};
+
+#define IST3931_INIT(inst)                                                                         \
+	static const struct ist3931_config ist3931_config_##inst = {                               \
+		.bus = I2C_DT_SPEC_INST_GET(inst),                                                 \
+		.reset_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, reset_gpios, {}),                     \
```

Comment:
```suggestion
		.reset_gpio = GPIO_DT_SPEC_INST_GET(inst, reset_gpios),                     \
```

(the `reset-gpios` property is `required: true`)